spec/measure_repository_service_test_kit/library_data_requirements_spec.rb

```diff
@@ -0,0 +1,299 @@
+    end
+
+    it 'fails if a 200 is not returned' do
+      stub_request(
+        :post,
+        "#{url}/Library/#{library_id}/$data-requirements"
+      ).to_return(status: 400, body: dr_library.to_json)
+
+      result = run(test, url:, library_id:, library_url:, library_identifier:, library_version:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if body is not a Library' do
+      stub_request(
+        :post,
+        "#{url}/Library/#{library_id}/$data-requirements"
+      ).to_return(status: 200, body: FHIR::Bundle.new(id: 'bundle_id').to_json)
+
+      result = run(test, url:, library_id:, library_url:, library_identifier:, library_version:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if Library in returned bundle is not of type module-definition' do
+      other_library = dr_library.clone
+      other_library.type.coding[0].code = 'logic-library'
+      stub_request(
+        :post,
+        "#{url}/Library/#{library_id}/$data-requirements"
+      ).to_return(status: 200, body: other_library.to_json)
+
+      result = run(test, url:, library_id:, library_url:, library_identifier:, library_version:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server returns 404 when no library matches id' do
+    let(:test) { group.tests[4] }
+
+    it 'passses when 404 returned with OperationOutcome' do
+      stub_request(
+        :post,
+        "#{url}/Library/INVALID_ID/$data-requirements"
+      ).to_return(status: 404, body: error_outcome.to_json)
+
+      result = run(test, url:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if 200 status code returned' do
+      stub_request(
+        :post,
+        "#{url}/Library/INVALID_ID/$data-requirements"
+      ).to_return(status: 200, body: error_outcome.to_json)
+
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if library returned' do
+      library = FHIR::Library.new
+      stub_request(
+        :post,
+        "#{url}/Library/INVALID_ID/$data-requirements"
+      ).to_return(status: 200, body: library.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+  end
+
+  describe 'Server returns 400 when no id, url, or identifier provided' do
+    let(:test) { group.tests[5] }
+
+    it 'passes when 400 returned with OperationOutcome' do
+      stub_request(
+        :post,
+        "#{url}/Library/$data-requirements"
+      ).to_return(status: 400, body: error_outcome.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if 200 status code returned' do
+      stub_request(
+        :post,
+        "#{url}/Library/$data-requirements"
+      ).to_return(status: 200, body: error_outcome.to_json)
+      result = run(test, url:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if library body returned' do
```

Comment:
Unsure if this is necessary, but one of the tests for a similar describe in `measure_data_requirements.rb` checks `fails if server does not return OperationOutcome`. Should we add that here?

Reply:
I think that test tests the same thing as this one, right? Looks to me like the only difference is the one for measure data requirements returns a 200 instead of 400 (which doesn't really matter since we just care about the response body)